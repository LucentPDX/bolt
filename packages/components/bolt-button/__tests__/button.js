const { render } = require('@bolt/twig-renderer');
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../button.schema.yml'));
const { tag } = schema.properties;

describe('button', async () => {
  test('basic button', async () => {
    const results = await render('@bolt-components-button/button.twig', {
      text: 'Hello World',
    });
    // console.log({ results });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  tag.enum.forEach(async tagChoice => {
    test(`button tag: ${tagChoice}`, async () => {
      const results = await render('@bolt-components-button/button.twig', {
        text: 'Hello World',
        tag: tagChoice,
      });
      // console.log({ results });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
