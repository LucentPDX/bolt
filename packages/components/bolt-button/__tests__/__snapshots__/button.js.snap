// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button basic button 1`] = `
"<bolt-button
   align=\\"center\\"    color=\\"primary\\"    size=\\"medium\\"            transform=\\"none\\"     
  
      
  
  >

    <button  class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary  c-bolt-button--center \\" is=\\"shadow-root\\">
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
    <replace-with-children class=\\"c-bolt-button__item \\">Hello World</replace-with-children>
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
  </button>
</bolt-button>"
`;

exports[`button button tag: a 1`] = `
"<bolt-button
   align=\\"center\\"    color=\\"primary\\"    size=\\"medium\\"            transform=\\"none\\"     
  
      
  
  >

    <a  class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary  c-bolt-button--center \\" is=\\"shadow-root\\">
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
    <replace-with-children class=\\"c-bolt-button__item \\">Hello World</replace-with-children>
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
  </a>
</bolt-button>"
`;

exports[`button button tag: button 1`] = `
"<bolt-button
   align=\\"center\\"    color=\\"primary\\"    size=\\"medium\\"            transform=\\"none\\"     
  
      
  
  >

    <button  class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary  c-bolt-button--center \\" is=\\"shadow-root\\">
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
    <replace-with-children class=\\"c-bolt-button__item \\">Hello World</replace-with-children>
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
  </button>
</bolt-button>"
`;

exports[`button button tag: link 1`] = `
"<bolt-button
   align=\\"center\\"    color=\\"primary\\"    size=\\"medium\\"            transform=\\"none\\"     
  
      
  
  >

    <a  class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary  c-bolt-button--center \\" is=\\"shadow-root\\">
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
    <replace-with-children class=\\"c-bolt-button__item \\">Hello World</replace-with-children>
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
  </a>
</bolt-button>"
`;

exports[`button button tag: reset 1`] = `
"<bolt-button
   align=\\"center\\"    color=\\"primary\\"    size=\\"medium\\"            transform=\\"none\\"     
  
      
   type=\\"reset\\"
  >

    <button  type=\\"reset\\" class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary  c-bolt-button--center \\" is=\\"shadow-root\\">
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
    <replace-with-children class=\\"c-bolt-button__item \\">Hello World</replace-with-children>
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
  </button>
</bolt-button>"
`;

exports[`button button tag: submit 1`] = `
"<bolt-button
   align=\\"center\\"    color=\\"primary\\"    size=\\"medium\\"            transform=\\"none\\"     
  
      
   type=\\"submit\\"
  >

    <button  type=\\"submit\\" class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary  c-bolt-button--center \\" is=\\"shadow-root\\">
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
    <replace-with-children class=\\"c-bolt-button__item \\">Hello World</replace-with-children>
          <replace-with-children class=\\"c-bolt-button__icon is-empty\\"></replace-with-children>
  
  </button>
</bolt-button>"
`;
